tests/unit_tests/c_wrappers/res/config/test_config_content.py

```diff
@@ -74,9 +112,35 @@ def test_config_content_as_dict(tmpdir):
             ConfigKeys.NUM_REALIZATIONS: 42,
             ConfigKeys.QUEUE_OPTION: [["SLURM", "MAX_RUNNING", "50"]],
             "MULTIPLE_KEY_VALUE": [6, 24, 12],
-            ConfigKeys.DATA_FILE: [
-                str(Path.cwd() / existing_file_2),
-                str(Path.cwd() / existing_file_1),
-            ],
+            ConfigKeys.DATA_FILE: str(Path.cwd() / existing_file_2),
             ConfigKeys.REFCASE: str(Path.cwd() / existing_file_1),
         }
+
+
+def test_config_content_as_dict_single_value_keys(tmpdir):
+    with tmpdir.as_cwd():
+        conf = ConfigParser()
+        existing_file_1 = "test_1.t"
+        existing_file_2 = "test_2.t"
+        Path(existing_file_2).write_text("something")
+        Path(existing_file_1).write_text("not important")
+        init_user_config_parser(conf)
+        type_value_map = {
+            "str": "abc",
+            "path": tmpdir,
+            "file_path": existing_file_1,
+            "bool": "TRUE",
+            "float": 4.2,
+            "int": 2,
+        }
+
+        with open("config.file", "w") as fileH:
+            for key in SINGLE_VALUE_KEYS:
+                key_type = SINGLE_VALUE_KEY_TYPES[key]
+                fileH.write(f"{key} {type_value_map.get(key_type, key_type)}\n")
+                fileH.write(f"{key} {type_value_map.get(key_type, key_type)}\n")
```

Comment:
the default value in the call to `get` is given in case an unexpected key (or rather a key with unexpected type) appears?

I'm sorry, maybe I'm a bit slow, but I don't quite understand parts of this test - one aspect is rather clear, the double entry that only yields a single value, as conveyed by the assert, but the necessity of the type value map is not immediately clear to me - is this purely technical, or does this also test something?

Reply:
OK i think i get it now - it's also a short way of supplying (valid according to type) values for all single value config keys. thanks for the explanation!